### PR TITLE
apt-get --reinstall option added

### PR DESCRIPTION
**Pull request by ndpu**
We need support to generate signatures of already installed packages. This commit added just this feature - using apt-get --reinstall option. Maybe it will be useful to someone else.

### CODE REVIEW

File: apt_offline_core/AptOfflineCoreLib.py [hunk 1]
```diff
@@ -1908,6 +1918,9 @@ def main():
         parser_set.add_argument("--install-src-packages", dest="set_install_src_packages", help="Source Packages that need to be installed",
                           action="store", type=str, nargs='*', metavar="SOURCE PKG")
         
+        parser_set.add_argument("--aptreinstall", dest="set_aptreinstall", help="Generate signatures of packages that already installed in system",
```
Comment (rickysarraf) [inline apt_offline_core/AptOfflineCoreLib.py]: Please make this more human readable. It anyway is a long option, thus --apt-reinstall will not hurt.

File: apt_offline_core/AptOfflineCoreLib.py [hunk 2]
```diff
@@ -1758,12 +1760,20 @@ def __AptInstallPackage(self, PackageList=None, ReleaseType=None):
...
                                 #FIXME: Find a more Pythonic implementation
-                                if self.__ExecSystemCmd( '/usr/bin/apt-get -qq --print-uris install $__apt_set_install_packages >> $__apt_set_install' ) is False:
-                                        log.err( "FATAL: Something is wrong with the apt system.\n" )
+                                cmd = '/usr/bin/apt-get -qq --print-uris %s install $__apt_set_install_packages >> $__apt_set_install'
+
+                        reinstall = ''
+                        if self.AptReinstall:
+                                reinstall = '--reinstall'
+
+                        cmd = cmd % reinstall
```
Comment (rickysarraf) [inline apt_offline_core/AptOfflineCoreLib.py]: Is this a typo ?

**Comment by rickysarraf** [comment]
And have you verified what happens in the 'install' target ?

Since the debs will already be present, does shutil.copy play nice ? Does it quietly overwrite the file to the destination ?

File: apt_offline_core/AptOfflineCoreLib.py (hunk 1 above)
Comment (ndpu) [inline apt_offline_core/AptOfflineCoreLib.py]: done

**Comment by ndpu** [comment]
>And have you verified what happens in the 'install' target ?
>Since the debs will already be present, does shutil.copy play nice ? Does it quietly overwrite the file to the destination ?

I've disabled using --release with --apt-reinstall if that's whats you meant (with target release is much harder to get all dependencies of installed packages by using only apt-get).

**Comment by rickysarraf** [comment]
On Tuesday 16 June 2015 10:59 PM, ndpu wrote:
>
>     And have you verified what happens in the 'install' target ?
>     Since the debs will already be present, does shutil.copy play nice
>     ? Does it quietly overwrite the file to the destination ?
>
> I've disabled using --release with --apt-reinstall if that's whats you
> meant (with target release is much harder to get all dependencies of
> installed packages by using only apt-get).
>

No. I mean what happens in the "install" operation.

But you can ignore this now. Because If you are requesting apt for a
reinstall, and if the pkg is in the local cache, it won't reflect in the
downloadable list.


What about the other comment ? The one where you had "cmd % '--reinstall' "

-- 
Ritesh Raj Sarraf
RESEARCHUT - http://www.researchut.com
"Necessity is the mother of invention."

File: apt_offline_core/AptOfflineCoreLib.py (hunk 2 above)
Comment (ndpu) [inline apt_offline_core/AptOfflineCoreLib.py]: No, just string formatting (i've added %s to command line string, i think you have'nt noticed this)
```python
cmd = 'apt-get %s install some_package'
reinstall = ''
cmd = cmd % reinstall
# cmd = 'apt-get   install some_package'
reinstall = '--reinstall'
cmd = cmd % reinstall
# cmd = 'apt-get --reinstall install some_package'
```

**Comment by rickysarraf** [comment]
I have had to revert your changes because the test cases failed. And the logs complained about your changes. Can you have a look at the Travis CI build ?

**Comment by ndpu** [comment]
> I have had to revert your changes because the test cases failed. And the logs complained about your changes. Can you have a look at the Travis CI build ?

fixed

**Comment by rickysarraf** [comment]
Yes. Please.

**Pull request merged**